### PR TITLE
[777] Migrate the find feature flags

### PR DESCRIPTION
**Pull request by darokel**
### Context

https://trello.com/c/va70CeOB/777-migrate-the-feature-flags

### Changes proposed in this pull request

- Migrates the feature flags logic from Find including related/dependent code

### Tech notes

Find uses Redis to store the feature flags. We'll need to decide whether to adopt it for all flags for both Find/Publish later. There's also a notification job to alert when flags are updated to Slack i've only set this to run in prod to avoid noise. It was only setup in prod in Find as well.

### Guidance to review

https://publish-teacher-training-pr-3108.london.cloudapps.digital/find/feature-flags

Use the normal publish credentials and toggle some features

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by darokel** [comment]
[Migrate the Feature Flags](https://trello.com/c/va70CeOB/777-migrate-the-feature-flags)

File: app/controllers/find/feature_flags_controller.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      FeatureFlag.send(action, feature_name)
+
+      if Rails.env.production?
+        SlackNotificationJob.perform_now(
+          ":flags: Feature ‘#{feature_name}‘ was #{action}d",
+          find_feature_flags_path,
+        )
+      end
+
+      flash[:success] = "Feature ‘#{feature_name.humanize}’ #{action}d"
```
Comment (defong) [inline app/controllers/find/feature_flags_controller.rb:18]: This don't get displayed in the right place, regression.

**Review by defong — CHANGES_REQUESTED** [review]

File: app/controllers/find/feature_flags_controller.rb (hunk 1 above)
Comment (darokel) [reply app/controllers/find/feature_flags_controller.rb:18]: Fixed

Screenshot 2022-11-09 at 14 32 10

File: app/controllers/find/feature_flags_controller.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+module Find
+  class FeatureFlagsController < ApplicationController
+    before_action :enforce_basic_auth
+    skip_before_action :redirect_to_maintenance_page_if_flag_is_active
```
[the remark this answers is not in the document]
Comment (darokel) [reply app/controllers/find/feature_flags_controller.rb:4]: Have moved this stuff over too https://github.com/DFE-Digital/publish-teacher-training/pull/3108/commits/25e85f2d07861ff0f6cdf1952c42d383fb792605

**Review by defong — COMMENTED** [review]
:dart: :foot: :boom: 

it a dead loop

**Comment by darokel** [comment]
> 🎯 🦶 💥
> 
> it a dead loop [ it a dead loop ](https://user-images.githubusercontent.com/470137/201082641-82edc252-8cb2-40f2-b9df-9e62782b4815.gif) [ ](https://user-images.githubusercontent.com/470137/201082641-82edc252-8cb2-40f2-b9df-9e62782b4815.gif)

😂

**Comment by darokel** [comment]
@defong should be fixed now

**Review by defong — COMMENTED** [review]
This don't seems to work :face_with_spiral_eyes: 

image

**Pull request merged**